### PR TITLE
Upgrade to Python 3.12

### LINKED ISSUE #39: Upgrade to Python 3.12
The CI, Docker Images etc should all be upgraded to Python 3.12.

### PR DESCRIPTION
**Pull request by VKTB**
## Description
Updates the CI jobs, Docker images, and pylint file to use Python 3.12.

## Testing instructions
- [ ] Review code
- [ ] Check Actions build
- [ ] Check app runs fine

## Agile board tracking
closes #39

### CODE REVIEW

File: .github/workflows/.ci.yml [hunk 1]
```diff
@@ -67,31 +67,31 @@ jobs:
   docker:
     # This job triggers only if all the other jobs succeed. It builds the Docker image and if successful,
     # it pushes it to Harbor.
-    needs: [linting, unit-tests]
+    needs: [ linting, unit-tests ]
```
Comment (joelvdavies) [inline .github/workflows/.ci.yml:70]: Same thing as https://github.com/ral-facilities/inventory-management-system-api/pull/201.

File: README.md [hunk 2]
```diff
@@ -8,8 +8,8 @@ This microservice requires an LDAP server to run against.
 
 ### Prerequisites
 - Docker and Docker Compose installed (if you want to run the microservice inside Docker)
+- Python 3.12 installed on your machine (if you are not using Docker)
```
Comment (joelvdavies) [inline README.md:11]: As in https://github.com/ral-facilities/inventory-management-system-api/pull/202/files, should the requires-python in the pyproject.toml be updated as well if we are saying its a requirement in the readme?

**Review by joelvdavies — COMMENTED** [review]
Worked locally, two minor comments.

File: README.md (hunk 2 above)
Comment (VKTB) [reply README.md:11]: Ah yes, I completely forgot about this, apologies..

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**